src/App.tsx

```diff
@@ -4,32 +4,71 @@ import logo from "../public/FischLogo.svg";
 
 import styles from "./App.module.css";
 import { Card } from "./components/Card";
-import { useRef, useState } from "react";
+import { JSX, useEffect, useState } from "react";
+import { NewCard } from "./components/NewCard";
 
-function App() {
-  const [cardsAmount, setCardsAmount] = useState(1)
-  const [cards, setCards] = useState([{}]);
+interface AppProps {
+  basicCards: {
+    face: string;
+    back: string;
+  }[];
+}
+
+interface CardObject {
+  face: string;
+  back: string;
+}
+
+function App(props: AppProps) {
+  const importedCards = props.basicCards.map((card, index) => {
+    return { face: card.face, back: card.back, key: index };
+  });
+
+  const [cardsArray, setCardsArray] = useState(importedCards);
```

Comment:
**nitpick:** It should look like following:

```jsx
const [cardsArray, setCardsArray] = useState<CardObject[]>([]);
```

Reply:
Then what am I gonna do with cards that I import from elsewhere? In previous iteration, when I was populating the final array (it was an array of JSX' then) from the imported cards, you said it was also wrong.

Reply:
You will use `useEffect` to populate fetched data to this array. You don't need props